flow/raster_cache.cc

```diff
@@ -24,13 +24,20 @@ RasterCacheResult::RasterCacheResult(sk_sp<SkImage> image,
 void RasterCacheResult::draw(SkCanvas& canvas, const SkPaint* paint) const {
   TRACE_EVENT0("flutter", "RasterCacheResult::draw");
   SkAutoCanvasRestore auto_restore(&canvas, true);
+
+// Sanity check that our raster cache size is close to the drawing bounds.
+#ifndef NDEBUG
   SkIRect bounds =
       RasterCache::GetDeviceBounds(logical_rect_, canvas.getTotalMatrix());
   FML_DCHECK(
       std::abs(bounds.size().width() - image_->dimensions().width()) <= 1 &&
       std::abs(bounds.size().height() - image_->dimensions().height()) <= 1);
+#endif
+
+  SkPoint offset =
+      canvas.getTotalMatrix().mapXY(logical_rect_.fLeft, logical_rect_.fTop);
```

Comment:
Does this work in a rotated coordinate system? The logical UL doesn't always transform to be the device space UL. Also, at 45 degree rotations, the coordinates where the image goes don't match any coordinate in the logical rect. I think bounds.UL was a better match for the correct position of where to paint the cached rectangle.

Reply:
Good catch! Fixed.